streams/src/main/java/org/apache/kafka/streams/kstream/internals/StreamJoinedStoreFactory.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.EmitStrategy;
+import org.apache.kafka.streams.kstream.JoinWindows;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.internals.StoreFactory;
+import org.apache.kafka.streams.state.DslWindowParams;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
+import org.apache.kafka.streams.state.WindowStore;
+
+public class StreamJoinedStoreFactory<K, V1, V2> extends AbstractConfigurableStoreFactory {
+
+    private final String name;
+    private final JoinWindows windows;
+    private final Serde<?> valueSerde;
+    private final WindowBytesStoreSupplier storeSupplier;
+    private final StreamJoinedInternal<K, V1, V2> joinedInternal;
+
+    private boolean loggingEnabled;
+    private final Map<String, String> logConfig;
+
+    public enum Type {
+        THIS,
+        OTHER
+    }
+
+    public StreamJoinedStoreFactory(
+            final String name,
+            final JoinWindows windows,
+            final StreamJoinedInternal<K, V1, V2> joinedInternal,
+            final Type type
+    ) {
+        super(joinedInternal.dslStoreSuppliers());
+        this.name = name + "-store";
+        this.joinedInternal = joinedInternal;
+        this.windows = windows;
+        this.loggingEnabled = joinedInternal.loggingEnabled();
+        this.logConfig = new HashMap<>(joinedInternal.logConfig());
+
+        // since this store is configured to retain duplicates, we should
+        // not compact, so we override the configuration to make sure that
+        // it's just delete (window stores are configured to compact,delete)
+        this.logConfig.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_DELETE);
```

Comment:
Is this new logic? I don't think it hurts anything to have compaction enabled...of course it's also not doing anything so I guess this optimization makes sense

Reply:
no, it's not new logic -- it was respected in a roundabout way earlier that isn't respected after the refactoring of it into a StoreFactory

Reply:
Also see the TODO i added on StoreFactory and LMKWYT

Reply:
I guess it would be nice to have a test for this logic but honestly that could probably be saved for the followup work mentioned in that TODO which would be a more involved refactoring of the changelog topic configuration.

(Haven't gotten to the tests yet so maybe you already did this)

Reply:
There is a test for it (not added in this PR)! that's how it caught this bug ;) I don't remember which file it's in though.. if I figure it out I'll link it here